### ISSUE TITLE
Separate examples from tests

### ISSUE THREAD

**Issue by Blokje5** [open-policy-agent/conftest#390]
Right now we often use the examples section both in the go tests and bats tests. However, we should strive to create realistic examples. I think it would be best to start separating the examples from the tests. This means we can start working on creating more realistic examples without worrying about breaking tests.

**Comment by jpreese** [comment]
100% agree with the idea. I do think that the examples still need to be tested to make sure that they are actually _working_ examples. But this lets us have more freedom about what we're testing without needing to worry about if its a good example and makes examples themselves easier to read.

**Comment by xchapter7x** [comment]
Im looking to get back into conftest work. I'd be happy to take a crack at refactoring some of our tests. 

Let me try to frame up the goals here:
1. config files in examples should not be used as fixtures in any of our tests
2. config files in examples should be tested for validity, such that they dont fall out of date

I think i'd break it out into 2 separate PRs for each of these goals. 

---

## goal 1:

```gherkin
given: a go test or a bats test
when: the test consumes a fixture config file
then: none of those fixture files are sourced from the same pool as the config examples set
```

## goal 2:

```gherkin
given: a set of example config files and conftest test suite
when: the conftest test suite is run
then: we evaluate the validity of the config files in examples directory
and either fail or pass depending on the state of the files. 
```

## Questions:

### goal 2:
  We might need to define what is a valid example config and what is not. 
  does valid mean, 
  A) we can parse the config files and the rego files
  or
  B) we see an expected outcome with regards to the policy evaluation


Thoughts @Blokje5  @jpreese ???

**Comment by jpreese** [comment]
Resolved via #530
